cloudformation/alarms.yaml

```diff
@@ -491,3 +491,93 @@ Resources:
       Period: 60
       Statistic: Maximum
       Threshold: 0
+  # cloudwatch synthetic support
+  MetadataCanary:
+    Type: AWS::Synthetics::Canary
+    Properties:
+      ArtifactS3Location: !Join ['', ['s3://', !Ref SyntheticsBucket, '/', !Join ['', [!Ref Stage, ig, canary]]]]
+      Code:
+        Handler: 'index.handler'
+        Script: |
+          var synthetics = require('Synthetics');
+          const log = require('SyntheticsLogger');
+
+          const apiCanaryBlueprint = async function () {
+    
+            // Handle validation for positive scenario
+            const validateSuccessful = async function(res) {
+              return new Promise((resolve, reject) => {
+                  if (res.statusCode < 200 || res.statusCode > 299) {
+                      throw res.statusCode + ' ' + res.statusMessage;
+                  }
+          
+                  let responseBody = '';
+                  res.on('data', (d) => {
+                      responseBody += d;
+                  });
+          
+                  res.on('end', () => {
+                      // Add validation on 'responseBody' here if required.
+                      resolve();
+                  });
+              });
+            };
+    
+
+            // Set request option for Verify /metadata
+            let path = "/" + process.env.STAGE + "/metadata"
+            let requestOptionsStep1 = {
+                hostname: process.env.HOST_NAME,
+                method: 'GET',
+                path: path,
+                port: '443',
+                protocol: 'https:',
+                body: "",
+                headers: {}
+            };
+            requestOptionsStep1['headers']['User-Agent'] = [synthetics.getCanaryUserAgentString(), requestOptionsStep1['headers']['User-Agent']].join(' ');
+
+            // Set step config option for Verify /metadata
+            let stepConfig1 = {
+                  includeRequestHeaders: false,
+                  includeResponseHeaders: false,
+                  includeRequestBody: false,
+                  includeResponseBody: false,
+                  restrictedHeaders: [],
+                  continueOnHttpStepFailure: false
+              };
+
+              await synthetics.executeHttpStep('Verify /metadata', requestOptionsStep1, validateSuccessful, stepConfig1);              
+          };
+
+          exports.handler = async () => {
+              return await apiCanaryBlueprint();
+          };
+      ExecutionRoleArn: !GetAtt SyntheticsRole.Arn

```

Comment:
What does ig mean here?

Reply:
IG is short for ImplementationGuide. I was trying to think of something shorter than metadata because the name got quite long.

Reply:
I would suggest something like 'metadataPingFailAlarm', since the metadata endpoint is the first thing to check if this alarm does go off. IG is a bit misleading here.

Reply:
The Name is limited to 21 characters so including the longest standard stage, prod, we'd be well over the max name limit. We could switch to `$stage-metadata-ping` which would be within the limit.

Reply:
pushed the name change to $stage-metadata-ping